### DISCUSSION TITLE
How to use Dashboard Variables with the CSV datasource

### DISCUSSION THREAD

**Discussion by srdennis03** [grafana/grafana-csv-datasource#228] [Q&A]
Hi,

The documentation for the Grafana CSV datasource states that Dashboard variables can be used Parameter and Header Key / Values pairings. But I cannot get his work at all. 

Are there any examples that show how the use of Dashboard Variables can filter / reduce the CSV data set ?

**Comment by DirtyOptics** [comment]
Agree, I have searched high and low for examples of using variables with no luck.

**Comment by DirtyOptics** [comment]
I create a new variable in my dashboard that extracts the field from my CSV 'SSID'. This variable grabs all the SSID's listed in my CSV. I can see this is working in the Variable options at the bottom.

In my dashboard, I try the following syntax in the 'field' name to call that variable.

$SSID
${SSID}
'SSID'
${SSID:csv}
filter = ${SSID}
filter = '${SSID}'
[[SSID:csv]]

But these do not seem to change the data. The variable has a tick that says the variable is referenced in the Dashboard.
Any ideas what I am doing wrong? I have read the wiki and Grafana docs, but can not find any suitable examples.

**Comment by zoltanbedi** [comment]
I can the variable interpolated when I use it with the following format `$title` check the query inspector to figure if it is not working for you @deeeblack. Also @srdennis03 this won't filter your data it will only include that column. I advise you to migrate to https://grafana.com/docs/plugins/yesoreyeram-infinity-datasource/latest/ to be able to filter your data as well.

**Reply by DirtyOptics** [reply]
This is how many threads remain unanswered for how to create a variable using the CSV plugin.

https://community.grafana.com/t/csv-variables-in-grafana/107916
https://community.grafana.com/t/variables-csv-expansion-in-dropdown/11436
https://community.grafana.com/t/using-variables-with-csv-datasource/73648
https://community.grafana.com/t/link-csv-values-of-variables-to-make-filtrations-in-the-data-panels/80466

All im after is some examples of the syntax because the documentation at Grafana does not seem to work using this plugin.
I am happy to help update your doco if required? Got to get it working first.

When I get this going I will document it with pictures.
